tests/test_transform.py

```diff
@@ -0,0 +1,231 @@
+        ("Leeder.04_30_2023.csv", "leeder", "2023-04-30"),
+        ("Magnum.30-09-2022.csv", "magnum", "2022-09-30"),
+        (
+            "mend-report Wallington.30_04_2023.csv",
+            "mend_report_wallington",
+            "2023-04-30",
+        ),
+        ("Report-of-Gohen.02-28-2023.csv", "report_of_gohen", "2023-02-28"),
+        ("rpt-Catalysm.2023-03-31.csv", "rpt_catalysm", "2023-03-31"),
+        ("TT_monthly_Trustmind.20220930.csv", "tt_monthly_trustmind", "2022-09-30"),
+        ("Virtous.05-31-2023 - securities.csv", "virtous", "2023-05-31"),
+        ("NoDateFile.csv", "no_date_file", None),
+        ("InvalidDate.99-99-9999.csv", "invalid_date", None),
+    ],
+)
+def test_extract_table_name_and_date(
+    temp_directories,
+    filename: str,
+    expected_table_name: Optional[str],
+    expected_date: Optional[str],
+):
+    """
+    Test the Transform.extract_table_name and ETLUtils.extract_date functions with various filename patterns.
+    Ensures table names are in snake_case and dates are correctly extracted.
+    """
+    input_dir, output_dir = temp_directories
+    config = Config(input_directory=input_dir, output_directory=output_dir)
+
+    # Extract table name using Transform class
+    extracted_table_name = ETLUtils.extract_table_name(filename)
+    assert (
+        extracted_table_name == expected_table_name
+    ), f"Failed table name extraction for filename: {filename}"
+
+    # Extract date using ETLUtils class
+    extracted_date = ETLUtils.extract_date(
+        filename, config.date_patterns, config.date_format
+    )
+    assert (
+        extracted_date == expected_date
+    ), f"Failed date extraction for filename: {filename}"
+
+
+def test_clean_csv_data(temp_directories, sample_csv_content):
+    """
+    Test the Transform.clean_csv_data function to ensure DATA_DATE is appended correctly,
+    columns are renamed to snake_case and uppercase, 'SEDOL' and 'ISIN' are renamed to 'INST_ID',
+    and 'SOURCE' column is added with correct table name.
+    """
+    input_dir, output_dir = temp_directories
+    filename = "Applebead.30-06-2023 breakdown.csv"
+    file_path = input_dir / filename
+    date = "2023-06-30"
+
+    # Write sample CSV content to input file
+    file_path.write_text(sample_csv_content)
+
+    # Call the clean_csv_data function
+    Transform.clean_csv_data(
+        filename=filename,
+        file_path=str(file_path),
+        output_directory=str(output_dir),
+        date=date,
+    )
+
+    # Check that the output file exists
+    output_file = output_dir / filename
+    assert output_file.exists(), f"Output file {filename} was not created."
+
+    # Read the transformed CSV
+    df = pl.read_csv(output_file)
+
+    # Check for 'DATA_DATE' column with correct date
+    assert "DATA_DATE" in df.columns, "DATA_DATE column is missing."
+    assert all(df["DATA_DATE"] == date), "DATA_DATE values are incorrect."
```

Comment:
(pandas has various test/equality methods you might use rather than iterating with `all`)